framework/core/src/Foundation/Concerns/InteractsWithLaravel.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Foundation\Concerns;
+
+use Flarum\Locale\LocaleManager;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Str;
+
+/**
+ * Implementation of the Laravel Application Contract,
+ * for the sake of better integration with Laravel packages/ecosystem.
+ */
+trait InteractsWithLaravel
```

Comment:
A lot of this stuff feels familiar, didn't we have it written somewhere before?

Reply:
predates my time in flarum, but it might have implemented the laravel ApplicationContract at some point and then removed.